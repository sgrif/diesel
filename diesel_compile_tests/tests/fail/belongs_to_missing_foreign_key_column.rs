#[macro_use]
extern crate diesel;

struct Bar;

table! {
    foo {
        id -> Integer,
    }
}

#[derive(Associations)]
#[diesel(belongs_to(Bar))]
#[diesel(table_name = foo)]
struct Foo1 {
    bar_id: i32,
}

#[derive(Associations)]
#[diesel(belongs_to(Bar, foreign_key = bar_id))]
#[diesel(table_name = foo)]
struct Foo2 {
    bar_id: i32,
}

#[derive(Associations)]
#[diesel(belongs_to(Bar))]
#[diesel(table_name = foo)]
struct Foo3 {
    #[diesel(column_name = bar_id)]
    baz_id: i32,
}

#[derive(Associations)]
#[diesel(belongs_to(Bar, foreign_key = bar_id))]
#[diesel(table_name = foo)]
struct Foo4 {
    #[diesel(column_name = bar_id)]
    baz_id: i32,
}

fn main() {}
