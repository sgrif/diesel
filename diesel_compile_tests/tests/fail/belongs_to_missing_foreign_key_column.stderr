error[E0412]: cannot find type `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:13:21
   |
13 | #[diesel(belongs_to(Bar))]
   |                     ^^^ not found in `foo`

error[E0425]: cannot find value `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:13:21
   |
13 | #[diesel(belongs_to(Bar))]
   |                     ^^^ not found in `foo`

error[E0412]: cannot find type `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:20:40
   |
20 | #[diesel(belongs_to(Bar, foreign_key = bar_id))]
   |                                        ^^^^^^ not found in `foo`

error[E0425]: cannot find value `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:20:40
   |
20 | #[diesel(belongs_to(Bar, foreign_key = bar_id))]
   |                                        ^^^^^^ not found in `foo`

error[E0412]: cannot find type `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:27:21
   |
27 | #[diesel(belongs_to(Bar))]
   |                     ^^^ not found in `foo`

error[E0425]: cannot find value `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:27:21
   |
27 | #[diesel(belongs_to(Bar))]
   |                     ^^^ not found in `foo`

error[E0412]: cannot find type `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:35:40
   |
35 | #[diesel(belongs_to(Bar, foreign_key = bar_id))]
   |                                        ^^^^^^ not found in `foo`

error[E0425]: cannot find value `bar_id` in module `foo`
  --> $DIR/belongs_to_missing_foreign_key_column.rs:35:40
   |
35 | #[diesel(belongs_to(Bar, foreign_key = bar_id))]
   |                                        ^^^^^^ not found in `foo`
